frontera/contrib/backends/remote/codecs/msgpack.py

```diff
@@ -6,11 +6,29 @@
 from msgpack import packb, unpackb
 
 from frontera.core.codec import BaseDecoder, BaseEncoder
+import six
 from w3lib.util import to_native_str
 
 
 def _prepare_request_message(request):
-    return [request.url, request.method, request.headers, request.cookies, request.meta]
+    def serialize(obj):
+        """Recursively walk object's hierarchy."""
+        if isinstance(obj, (bool, six.integer_types, float, six.binary_type, six.text_type)):
+            return obj
+        elif isinstance(obj, dict):
+            obj = obj.copy()
+            for key in obj:
+                obj[key] = serialize(obj[key])
+            return obj
+        elif isinstance(obj, list):
+            return [serialize(item) for item in obj]
+        elif isinstance(obj, tuple):
+            return tuple(serialize([item for item in obj]))
+        elif hasattr(obj, '__dict__'):
+            return serialize(obj.__dict__)
+        else:
+            return None
```

Comment:
let's make it generate warning, that object of some non-serializable type was skipped.